evergreen.yml

```diff
@@ -41,276 +41,273 @@
-      args:
-      - -v
-      - resmoke-test
-      - --suites
-      - ../../src/genny/src/resmokeconfig/genny_sharded.yml
-
-# Must run this task last to avoid polluting test results with tests generated
-# by the new actor.
-- name: t_test_create_new_actor_script
-  commands:
-  - func: f_run_create_new_actor_script
-  - func: f_compile
-  - func: f_dry_run_new_actor_script  # Test the generated yaml file is valid.
-  - command: subprocess.exec
-    params:
-      binary: ./src/genny/run-genny
-      args:
-      - -v
-      - resmoke-test
-      - --create-new-actor-test-suite
+  - name: t_integration_test_single_node_replset
+    commands:
+      - command: subprocess.exec
+        params:
+          binary: ./src/genny/run-genny
+          args:
+            - -v
+            - resmoke-test
+            - --suites
+            - ../../src/genny/src/resmokeconfig/genny_single_node_replset.yml
+
+  - name: t_integration_test_three_node_replset
+    commands:
+      - command: subprocess.exec
+        params:
+          binary: ./src/genny/run-genny
+          args:
+            - -v
+            - resmoke-test
+            - --suites
+            - ../../src/genny/src/resmokeconfig/genny_three_node_replset.yml
+
+  - name: t_integration_test_sharded
+    commands:
+      - command: subprocess.exec
+        params:
+          binary: ./src/genny/run-genny
+          args:
+            - -v
+            - resmoke-test
+            - --suites
+            - ../../src/genny/src/resmokeconfig/genny_sharded.yml
+
+  # Must run this task last to avoid polluting test results with tests generated
+  # by the new actor.
+  - name: t_test_create_new_actor_script
+    commands:
+      - func: f_run_create_new_actor_script
+      - func: f_compile
+      - func: f_dry_run_new_actor_script # Test the generated yaml file is valid.
+      - command: subprocess.exec
+        params:
+          binary: ./src/genny/run-genny
+          args:
+            - -v
+            - resmoke-test
+            - --create-new-actor-test-suite
 
 ##                ⚡️ Task Groups ⚡️
 
 task_groups:
+  - &a_compile_and_test
+
+    name: tg_compile_and_test
+    max_hosts: 1
+    teardown_task:
+      # Attaching results is prohibited in "teardown_group".
+      # So we call it in "teardown_task" but make the result file optional.
+      - func: f_report_test_results
+    tasks:
+      # Rely on implicit dependencies in task_groups where tasks are executed
+      # in the order they're defined. Tasks with explicit "depends_on" tasks
+      # may conflict with task_group's implicit dependency, causing
+      # unexpected behavior.
+      #
+      # For this task group, the implicit dependency is that t_*_test has to
+      # run after t_compile_and_dry_run.
+      - t_compile_and_dry_run
+      - t_python_test
+      # - t_lint_python
+      - t_lint_workloads
+      - t_check_generated_docs
+      - t_cmake_test
+
+  - !!merge <<: *a_compile_and_test
```

Comment:
Should we should be using merge expansions. Do they work with out parser and / or are they standard?

Reply:
I believe it is standard. My test code written using PyYAML had no with it. That said, I'm going to run a full patch as you requested in an earlier comment.